### ISSUE TITLE
crash while rendering youtube-viewer results (most likely to be emoji)

### ISSUE THREAD

**Issue by hydrocodone-senpai** [LukeSmithxyz/st#37]
(this is my first github issue so pls tell me if i need to provide more information)

I use youtube-viewer to browse Luke's videos because I hate using bloated webkits to perform such a basic service. I'm using this fork of st bc i'm too much of a brainlet to get the patches working, and anyways youtube-viewer almost always causes st to crash. Running st in another terminal, I get the following output.

X Error of failed request:  BadLength (poly request too large or internal Xlib length error)
  Major opcode of failed request:  139 (RENDER)
  Minor opcode of failed request:  20 (RenderAddGlyphs)
  Serial number of failed request:  4621
  Current serial number in output stream:  4841

Interestingly, it seems to be dependent on window size. When the terminal is less than roughly 1/3 of my screen, youtuve-viewer renders fine. I hypothesize this is because the terminal size "cuts off" incompatible emoji titles in Luke Smith's videos, and prevents st from crashing. More interestingly, it seems only some emoji are affected. The "lightning bolts" render fine.

![20190101-114433](https://user-images.githubusercontent.com/44106031/50575852-9f590400-0dba-11e9-862b-e73e51ba2020.png)

So I changed the font to Unifont and it did render Luke Smith's video titles without crashing. However... it's output is a little less than ideal.

![20190101-115855](https://user-images.githubusercontent.com/44106031/50575955-a3862100-0dbc-11e9-8968-4cdb257b0b41.png)

I want to keep my Liberation (FREE AS IN FREEDUMS) Sans font, and I don't want to have to be limited by fonts that must support emoji unicodes. Is there a way for st to handle like URxvt and just "default" to a block when it fails to render a character? And if there isn't, can Luke pls include a statement in the README listing some cool, hip fonts that renders unicode emojis well in terminals? Thanks.

**Comment by hydrocodone-senpai** [comment]
I have no idea why, but it seems that changing the font from Unifont back to Liberation Mono in .Xresources removed the bug for me. I really have no idea why this happened, but since I can't reproduce this, I'm going to close this issue until I can encounter a reliable way to generate this error.

**Comment by hydrocodone-senpai** [comment]
Really hate to necrobump this, but I found the issue and I hope anybody experiencing a similar issue can get help here. I did not have my locale set. To see your current locale:
`% localectl`
I had my system locale set to C for some reason. To change this, just do
`% localectl set-locale LANG=[insert locale]`.
I set my locale to en_US.utf8, and restarted. Then all unicode in my terminal rendered properly.
